src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_connection/manage_direct_query_data_connections_table.tsx

```diff
@@ -3,145 +3,215 @@
+          i18n.translate('dataSourcesManagement.directQueryTable.deleteDataSourceFailMsg', {
+            defaultMessage:
+              'Error occurred while deleting selected records for Data sources. Please try it again',
+          })
+        );
       })
       .finally(() => {
-        setIsLoading(false);
+        setIsDeleting(false);
       });
-  }, [http, notifications.toasts, selectedDataSourceId, featureFlagStatus]);
+  };
 
-  const deleteDataSources = useCallback(
-    (connectionName: string | undefined) => {
-      if (!connectionName || (featureFlagStatus && selectedDataSourceId === undefined)) return;
+  /* render delete modal*/
+  const tableRenderDeleteModal = () => {
+    return isModalVisible ? (
+      <EuiConfirmModal
+        title={i18n.translate('dataSourcesManagement.directQueryTable.multiDeleteTitle', {
+          defaultMessage: 'Delete data source connection(s)',
+        })}
+        onCancel={() => {
+          setIsModalVisible(false);
+        }}
+        onConfirm={() => {
+          setIsModalVisible(false);
+          onClickDelete();
+        }}
+        cancelButtonText={i18n.translate('dataSourcesManagement.directQueryTable.cancel', {
+          defaultMessage: 'Cancel',
+        })}
+        confirmButtonText={i18n.translate('dataSourcesManagement.directQueryTable.delete', {
+          defaultMessage: 'Delete',
+        })}
+        buttonColor="danger"
+        defaultFocusedButton="confirm"
+      >
+        <p>
+          <FormattedMessage
+            id="dataSourcesManagement.directQueryTable.deleteDescription"
+            defaultMessage="This action will delete the selected data source connections"
+          />
+        </p>
+        <p>
+          <FormattedMessage
+            id="dataSourcesManagement.directQueryTable.deleteConfirmation"
+            defaultMessage="Any objects created using data from these sources, including Index Patterns, Visualizations, and Observability Panels, will be impacted."
+          />
+        </p>
+        <p>
+          <FormattedMessage
+            id="dataSourcesManagement.directQueryTable.deleteWarning"
+            defaultMessage="This action cannot be undone."
+          />
+        </p>
+      </EuiConfirmModal>
+    ) : null;
+  };
 
-      const endpoint =
-        featureFlagStatus && selectedDataSourceId !== undefined
-          ? `${DATACONNECTIONS_BASE}/${connectionName}/dataSourceMDSId=${selectedDataSourceId}`
-          : `${DATACONNECTIONS_BASE}/${connectionName}`;
+  const renderToolsLeft = useCallback(() => {
+    return selectedDataSources.length > 0 && !isModalVisible
+      ? [
+          <EuiSmallButton
+            color="danger"
+            onClick={() => setIsModalVisible(true)}
+            data-test-subj="deleteDataSourceConnections"
+          >
+            Delete {selectedDataSources.length || ''}{' '}
+            {selectedDataSources.length ? 'connection' : ''}
+            {selectedDataSources.length >= 2 ? 's' : ''}
```

Comment:
We have `selectedDataSources.length > 0` in the path.
```suggestion
            <FormattedMessage
              id="dataSourcesManagement.directQueryTable.deleteToolLabel"
              defaultMessage="{selectionSize, plural, one {Delete # connection} other {Delete # connections}}"
              values={{ selectionSize: selectedDataSources.length }}
            />
```